syncstorage/tests/functional/test_storage.py

```diff
@@ -802,15 +807,16 @@ def test_batch(self):
         self.assertEquals(len(res['success']), max_count)
         self.assertEquals(len(res['failed']), 5)
 
-        # The test config has max_bytes=1M.
-        # Uploading 5 210MB items should produce one failure.
-        max_bytes = get_limit_config(self.config, 'max_post_bytes')
-        self.assertEquals(max_bytes, 1024 * 1024)
-        bsos = [{'id': str(i), 'payload': "X" * (210 * 1024)}
-                for i in range(5)]
+        # Uploading N+1 210kB items should produce one failure.
+        item_size = (210 * 1024)
+        max_items = max_bytes / item_size
+        self.assertTrue(max_items * item_size < max_bytes)
+        self.assertTrue((max_items + 1) * item_size > max_bytes)
+        bsos = [{'id': str(i), 'payload': "X" * item_size}
+                for i in range(max_items + 1)]
         res = self.app.post_json(self.root + '/storage/col2', bsos)
         res = res.json
-        self.assertEquals(len(res['success']), 4)
+        self.assertEquals(len(res['success']), max_items)
         self.assertEquals(len(res['failed']), 1)
```

Comment:
Arguably, everything should fail inside a batch if a single record fails... (Normally the client could just not commit the batch, but that's not an option for batch=1234&commit=true where the body has records, which for desktop – and probably the other clients – it almost always will). But that's certainly a discussion for a different time...

I guess if we respect all the config limits, there should be no failures, so it probably doesn't matter.

Reply:
Yeah, I agree, and in fact I feel like it would be simpler all round if any bad item in a request just failed the whole request.  But I think we kept it this way just so we didn't have to mess with existing client code, and particularly so we didn't have different behaviour between batch and non-batch cases.